logger/options.go

```diff
@@ -37,50 +37,131 @@ type Options struct {
 	MaxBufferLen  int
 	Meta          string
 	Tags          string
+}
+
+// MessageOptions encapsulates user-provided options such as the Level and App
+// that are passed along with each log.
+type MessageOptions struct {
+	App           string
+	Env           string
+	FlushInterval time.Duration
+	SendTimeout   time.Duration
+	Hostname      string
+	IndexMeta     bool
+	IngestURL     string
+	IPAddress     string
+	Level         string
+	MacAddress    string
+	MaxBufferLen  int
+	Meta          string
+	Tags          string
 	Timestamp     time.Time
 }
```

Comment:
You might want to consider just embedding `Options` inside of `MessageOptions` to reuse it instead of duplicating it:

```golang
type MessageOptions struct {
    Options
    Timestamp time.Time
}
```

Reply:
I was thinking about doing it that way too- my thought was that it could be odd for a user (when initializing their options) to have timestamp be outside of the rest of the options.

Reply:
this is what I had in mind as well here. I think it's OK that it's slightly more verbose for an enduser who is doing overrides in a `LogWithOptions`, e.g:

```golang
logger.LogWithOptions("message", MessageOptions{Options{App: "anotherapp"}, time.Now()})
```